[Huang, Po-Wei] iP

georgepwhuang opened the pull request.

Commit:
Add Gradle support

Commit:
build.gradle: Update version to 8.29

Commit:
Implement greet, echo and exit

Project starts with a basic skeletal version.

Let's allow BLARB, which is this version of Duke to greet the users
and exit when called, and to echo all other unrecognized commands.

The skeletal version is to kick off the project without complications.

Commit:
Add comments and change variable names

Increase clarity on the functionality of the code.

Variable names are changed to fit the problem statement.

Commit:
Implement add and list

BLARB should store and list text inputted by the user.

Edit the echo utility into storing the inputted text.
Add listing utility to call stored text.

Such utility is used to later further implement other functions.

Commit:
Implement ability to mark tasks as done

Allow stored tasks to function as checklist.

Converted storage from Strings to self-defined immutable class Task.

This is so that the completion attribute is bundled with the task,
rather than separately denoted.

Commit:
Add support for different types of Tasks

There are different types of tasks and their properties
should be taken into account.

Created three subtypes of Task. Changed Task to a mutable
class and allow toggling of the completion status.
Parsing of the input strings by split() is implemented.

The immutable state of the Task class is undesirable
for inheritance. Parsing method is made to fit the
command-description style of input.

Commit:
Abstract chat functions to a separate class

The Duke class is messy and not written in an object-oriented style.

Abstracted all chatting functions to new class so that the Duke
class only does the logo printing and input looping.

The main class is made cleaner and the chatting functions are
rewritten in OOP style to clear the code and enhance readability.

Commit:
Implement shell script testing

Shell script testing is used to faster debug the code.

Created test case and fixed minor bugs in the output in order to
decrease issues in the shell script, namely, the spaces at the end
of output lines are deleted.

Spaces at the end of a line cause comparability issues, and causes
the shell script to fail test cases even if outputs are the same.

Commit:
Increase exception handling cases

The code should have more flexibility for the acceptance of exceptions.

Various try-catch blocks are added to catch different exception types
and to give different responses according to.

The use of Java inbuilt exceptions are used directly rather than self-
defined ones are due to the inbuilt exceptions being sufficient for
the current exceptions that may be caught.

Commit:
Add delete function

The user should be able to delete tasks that are no longer needed.

A delete function is added to the code. Some Integer types are changed
to int types.

The conversion of reference types to primitive types is implemented
in order allow easier deletions in ArrayLists.

Commit:
Convert commands to enum types

The commands in the Blarb class is implemented through a long
if-else if-else block, while switch-case blocks would allow cleaner
code.

The enum type for commands are created, and the long if-else if-else
block in Blarb.java is replaced with a switch-case.

The enum type is used in order to allow for the switch-case block to
be used as it only accepted single variable types.

Commit:
Add values to enum type Command and reformatted code

The Command class is still dependant on a long if-else is-else block.

String values are attached to each Command.
Cleaned code to fit Java coding standards.

Commands can be called directly from the input string rather than
going through a long selection block.

Commit:
Add store function

The user should be able to call the same task list across multiple
iterations of program execution.

file() and refile() functions are added to store the current task list
into an output file. The file data is also transferred into the task
list upon initialization of the program.

The program has the flexibility to either append to the storage file or
completely rewrite it, hence the two functions of file() and refile().

The main class is renamed as Main.java for the removal of Duke
references to the program.

Commit:
Add datetime support

Deadlines should be shown in a datetime format.

The deadline task initializations now only supports deadlines in the
form of yyyy-mm-dd.

This is to restrict the usage of random not time related deadlines.

Commit:
Merge branch 'branch-Level-7'

Commit:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Blarb.java

Commit:
Increase OOP extraction

OOP styled coding should be adopted to increase modularity.

The following classes have been extracted from Blarb:
Ui, Parser, Storage, and Tasklist.
Blarb still retains the main execution of commands parsed from the
Parser.

This is to prevent the Blarb class from growing too large.

Commit:
Package classes into blarb package

Software engineering projects should be appropriately packaged to ensure
that implementation details are not seen by the client.

All classes except Main are wrapped into the blarb package. Access
modifiers for packaged classes, except Blarb are set to
package-private to hide details and as the Main class only requires
the Blarb class to execute.

Most classes are set to package-private for proper encapsulation.

Commit:
Add JUnit tests

Classes in a project should be tested to check their workability.

JUnit tests are created for the Deadline class and Command class.

The two classes are chosen for their vitality and their independence as
a class.

Commit:
Package as JAR

Projects should be packaged into executables.

The project is packaged into a JAR file.

Commit:
Add missing Javadocs

Javadocs are added to induce clarity of the code.

Missing Javadocs of most methods are added and some existing Javadocs
are edited.

Commit:
Add find function

Users should be able to search through their tasklist.

The find function is implemented. Some bugs caused from further OOP
implementation are fixed.

Commit:
Tweak code to follow Java coding standard

Coding practices should follow coding standards.

Import statements and grouping are adjusted. Minor bugs fixed.

Commit:
Merge branch 'branch-A-JavaDoc'

Commit:
Merge branch 'branch-A-CodingStandard'

Commit:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/blarb/Tasklist.java

georgepwhuang committed:
Merge commit 'a75fceeb74708da6ac524c08ef526d38e60ee56b' into branch-A-Gradle

* commit 'a75fceeb74708da6ac524c08ef526d38e60ee56b':
  build.gradle: Update version to 8.29
  Add Gradle support

georgepwhuang committed:
Add Gradle Support

Gradle support added to execute project tasks via Gradle rather than manually.

Moved JUnit test files to suitable directory. Edited build.gradle file to fit project settings.

Enable multiple Gradle tasks including run, test, shadowJar...etc

georgepwhuang committed:
Merge branch 'branch-A-Gradle'

georgepwhuang committed:
Add Checkstyle Support

Checkstyle is used to clean up the code.

Edited code to fit coding standards better with Checkstyle.

georgepwhuang committed:
Merge branch 'branch-A-Checkstyle'

georgepwhuang committed:
Add GUI Support

GUIs are a much better user interface than text-based UIs, hence support
is added.

FXML styled graphic design is implemented.
Processing for Blarb is extracted from the main Blarb class to support
both text UI and GUI.
Gradle is updated to support GUI as well.

georgepwhuang committed:
Merge branch 'branch-Level-10'

georgepwhuang committed:
Add Varargs usage

Used Varargs for parsing.
Cleaned code and fixed minor bugs.
Opened nested exception handling blocks in the Processor class.

Enhance code readability.

georgepwhuang committed:
Merge branch 'branch-A-Varargs'

ming-00 commented on src/main/java/blarb/App.java:
```diff
@@ -0,0 +1,34 @@
+    public void start(Stage stage) {
+        Blarb blarb = new Blarb();
+        try {
+            FXMLLoader fxmlLoader = new FXMLLoader(App.class.getResource("/view/MainWindow.fxml"));
```
Indentation for wrapped lines should be 8 spaces (i.e. twice the normal indentation of 4 spaces) more than the parent line

ming-00 commented on src/main/java/blarb/Blarb.java:
```diff
@@ -0,0 +1,100 @@
+     * @param input The inputted command.
+     * @return A boolean value that shows the availability for the next command intake.
+     */
+    public boolean execute(String input) {
```
Boolean variables/methods should be named to sound like booleans

ming-00 commented on src/main/java/blarb/Blarb.java:
```diff
@@ -0,0 +1,100 @@
+     * @param input User input string.
+     * @return Response string.
+     */
+    public boolean leave(String input) {
```
Boolean variables/methods should be named to sound like booleans

ming-00 commented on src/main/java/blarb/Processor.java:
```diff
@@ -0,0 +1,145 @@
+                    : "Type \"bye\" to see me go.";
+            break;
+        default:
+            outputs[0] = String.format("I have detailed files on human anatomy, but not %s.", input);
```
Indentation for wrapped lines should be 8 spaces (i.e. twice the normal indentation of 4 spaces) more than the parent line

ming-00 reviewed (commented):
LGTM!

georgepwhuang committed:
Increase code quality

Increase the readability according to peer reviews and further
abstraction.

Abstracted processing into various self-contained functions to lower
nesting, and increase abstractions.

Changed boolean names to look more self-explanatory.

Fixed possible errors caused by JavaFX-versions.

Fixed minor bugs in tasklist editing.

georgepwhuang committed:
Enable Assertions

Assertions are added to the code to ensure that the code would not have
unexpected errors.

georgepwhuang committed:
Add Lambdas

Add a bit of Functional Programming to replace syntactic sugars.

Used the Optional class in combination with lambda functions to avoid
the usage and assignment of null values.

georgepwhuang committed:
Add Streams

Add a bit of Streams to replace syntactic sugars.

georgepwhuang committed:
Build Second Jar

Fixed Minor Bugs in task printing and edited build.gradle file for jar
publishing.

georgepwhuang committed:
Merge pull request #1 from georgepwhuang/branch-A-Assertions

Enable Assertions

georgepwhuang committed:
Merge branch 'master' into branch-A-CodeQuality

georgepwhuang committed:
Merge pull request #2 from georgepwhuang/branch-A-CodeQuality

Increase code quality

georgepwhuang committed:
Merge branch 'master' into branch-A-Lambdas

georgepwhuang committed:
Merge pull request #3 from georgepwhuang/branch-A-Lambdas

Add Lambdas

georgepwhuang committed:
Merge pull request #4 from georgepwhuang/branch-A-Streams

Add Streams

georgepwhuang committed:
Add CI Support

Use GitHub Actions to set up Continuous Integration

georgepwhuang committed:
Merge branch 'branch-A-CI'

georgepwhuang committed:
Add edit function

Allows the user to edit the previously stored tasks.

Fixed bugs caused by multi-way merging.

georgepwhuang committed:
Merge branch 'branch-BCD-Extension'

georgepwhuang committed:
Improved the GUI

The GUI should look nice and not irritating.

JavaFX CSS is implemented. Dialog boxes are padded and colored. Errors
are highlighted red.

georgepwhuang committed:
Merge branch 'branch-A-BetterGui'

georgepwhuang committed:
Add User Guide

A basic User Guide is added in the README.md file.

A screenshot of the UI is also uploaded.

georgepwhuang committed:
Set theme jekyll-theme-cayman

georgepwhuang committed:
Set theme jekyll-theme-slate

georgepwhuang committed:
Set theme jekyll-theme-slate
